src/blenderbim/blenderbim/bim/module/spreadsheet_writer/operator.py

```diff
@@ -0,0 +1,652 @@
+
+        for product in products:
+            ##################################################################
+            ##########################  GlobalId #############################
+            ##################################################################
+            ifc_dictionary['GlobalId'].append(str(product.GlobalId))
+             
+            ##################################################################
+            ########################### General ##############################
+            ##################################################################
+            if blenderbim_spreadsheet_properties.my_ifcproduct:
+                ifc_dictionary['IfcProduct'].append(str(product.is_a()))
+                
+            if blenderbim_spreadsheet_properties.my_ifcbuildingstorey:  
+                ifc_dictionary['IfcBuildingStorey'].append(self.get_ifcbuildingstorey(context, ifcproduct=product, ifc_version=ifc_version)[0])    
+    
+            if blenderbim_spreadsheet_properties.my_ifcproduct_name: 
+                ifc_dictionary['Name'].append(str(product.Name))
+                
+            if blenderbim_spreadsheet_properties.my_type:     
+                ifc_dictionary['Type'].append(self.get_ifcproducttype_name(context, ifcproduct=product)[0])
+            
+            if blenderbim_spreadsheet_properties.my_ifcclassification: 
+                ifc_dictionary['Classification'].append(self.get_classification_code(context, ifcproduct=product, ifc_version=ifc_version)[0])
+              
+            if blenderbim_spreadsheet_properties.my_ifcmaterial:     
+                ifc_dictionary['Material(s)'].append(self.get_materials(context, ifcproduct=product)[0])
+                
+            
+            ##################################################################
+            ################### Common Properties ############################
+            ##################################################################  
+            if blenderbim_spreadsheet_properties.my_isexternal:    
+                ifc_dictionary[is_external].append(self.get_common_properties(context, ifcproduct=product, property_name=is_external)[0])
+                
+            if blenderbim_spreadsheet_properties.my_loadbearing:     
+                ifc_dictionary[loadbearing].append(self.get_common_properties(context, ifcproduct=product, property_name=loadbearing)[0])
+                 
+            if blenderbim_spreadsheet_properties.my_firerating:    
+                ifc_dictionary[fire_rating].append(self.get_common_properties(context, ifcproduct=product, property_name=fire_rating)[0])
+               
+            ##################################################################
+            ##################### Base Quantities ############################
+            ##################################################################
+            if blenderbim_spreadsheet_properties.my_length:  
```

Comment:
Since the code is reused a lot it may be condensed a bit, for instance :
```
for quantity_name in (length, width, height):  # etc.
    if getattr(blenderbim_spreadsheet_properties, f"my_{quantity_name.lower()}"):
        ifc_dictionary[quantity_name].append(self.get_quantities(context, ifcproduct=product, quantity_name=quantity_name)[0])
```